spec/client_spec.rb

```diff
@@ -366,4 +370,65 @@ def send_stream_content(res, content, keep_open:)
       end
     end
   end
+
+  it "connects to HTTP server through proxy" do
+    events_body = simple_event_1_text
+    with_server do |server|
+      server.setup_response("/") do |req,res|
+        send_stream_content(res, events_body, keep_open: false)
+      end
+      with_server(StubProxyServer.new) do |proxy|
+        event_sink = Queue.new
+        client = subject.new(server.base_uri, proxy: proxy.base_uri) do |c|
+          c.on_event { |event| event_sink << event }
+        end
+
+        with_client(client) do |client|
+          expect(event_sink.pop).to eq(simple_event_1)
+          expect(proxy.request_count).to eq(1)
+        end
+      end
+    end
+  end
+
+  it "resets read timeout between events" do
+    event_body = simple_event_1_text
+    with_server do |server|
+      attempt = 0
+      correctly_closed = true
+      server.setup_response("/") do |req,res|
+        attempt += 1
+        if attempt == 1
+          stream = send_stream_content(res, event_body, keep_open: true)
+          Thread.new do
+            2.times {
+              # write within timeout interval
+              sleep(0.75)
+              stream.write(event_body)
+            }
+            # cause timeout
+            sleep(1.25)
+            # should never get here
+            correctly_closed = false
```

Comment:
Why would we never get here, and what would it mean if we did get here? There isn't anything that would interrupt this thread. It looks to me like you're counting on the part of the test that executes after the timeout happens to take less than 0.25 seconds, which is very likely to be true but I'm unclear on how it relates to the test logic.

Reply:
Ah you are right what I wrote does not make sense, and I am not sure what my logic was. I still want to check that the connection is actually closed by the client though.

Reply:
That'd be good, but I'm not sure if there's a way in Webrick (the HTTP server framework that's being used here) for a request handler to know whether the connection's been closed.

Reply:
I think there isn't, because their abstraction doesn't give you direct access to the socket.

Reply:
@eli-darkly I was not able to get detection of proper connection closure working.

Reply:
Alas